src/main/java/uk/gov/digital/ho/hocs/audit/auditdetails/repository/AuditRepository.java

```diff
@@ -1,15 +1,38 @@
 package uk.gov.digital.ho.hocs.audit.auditdetails.repository;
 
-import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.stereotype.Repository;
 import uk.gov.digital.ho.hocs.audit.auditdetails.model.AuditData;
 
+import java.time.LocalDateTime;
+import java.util.List;
 import java.util.UUID;
 
 @Repository
-public interface AuditRepository extends CrudRepository<AuditData, String> {
+public interface AuditRepository extends PagingAndSortingRepository<AuditData, String> {
 
-    AuditData findByUuid(UUID uuid);
 
+    AuditData findAuditDataByUuid(UUID uuid);
+
+    @Query(value = "SELECT a.* FROM audit_data a WHERE a.correlation_id = ?1 AND a.audit_timestamp > ?2", nativeQuery = true)
+    List<AuditData> findAuditDataByCorrelationID(String correlationID, LocalDateTime dateTime, Pageable pageRequest);
+
+    @Query(value = "SELECT a.* FROM audit_data a WHERE a.correlation_id = ?1 AND a.audit_timestamp > ?2 AND a.audit_timestamp < ?3", nativeQuery = true)
+    List<AuditData> findAuditDataByCorrelationIDAndDateRange(String correlationID, LocalDateTime dateFrom, LocalDateTime dateTo, Pageable pageRequest);
```

Comment:
I would always use the BETWEEN syntax here for comparing the timestamps, but that's preference.